specs/sharding/beacon-chain.md

```diff
@@ -103,786 +87,558 @@ The following values are (non-configurable) constants used throughout the specif
 
-This epoch transition overrides Bellatrix epoch transition:
+        # Verify the degree proof
+        r = hash_to_field(sharded_commitments_container.sharded_commitments)
+        r_powers = compute_powers(r, len(sharded_commitments_container.sharded_commitments))
+        combined_commitment = elliptic_curve_lincomb(sharded_commitments_container.sharded_commitments, r_powers)
 
-```python
-def process_epoch(state: BeaconState) -> None:
-    # Sharding pre-processing
-    process_pending_shard_confirmations(state)
-    reset_pending_shard_work(state)
-
-    # Base functionality
-    process_justification_and_finalization(state)
-    process_inactivity_updates(state)
-    process_rewards_and_penalties(state)  # Note: modified, see new TIMELY_SHARD_FLAG_INDEX
-    process_registry_updates(state)
-    process_slashings(state)
-    process_eth1_data_reset(state)
-    process_effective_balance_updates(state)
-    process_slashings_reset(state)
-    process_randao_mixes_reset(state)
-    process_historical_roots_update(state)
-    process_participation_flag_updates(state)
-    process_sync_committee_updates(state)
-```
+        verify_degree_proof(combined_commitment, SAMPLES_PER_BLOB * FIELD_ELEMENTS_PER_SAMPLE, sharded_commitments_container.degree_proof)
 
-#### `process_pending_shard_confirmations`
+        # Verify that the 2*N commitments lie on a degree N-1 polynomial
+        low_degree_check(sharded_commitments_container.sharded_commitments)
 
-```python
-def process_pending_shard_confirmations(state: BeaconState) -> None:
-    # Pending header processing applies to the previous epoch.
-    # Skip if `GENESIS_EPOCH` because no prior epoch to process.
-    if get_current_epoch(state) == GENESIS_EPOCH:
-        return
-
-    previous_epoch = get_previous_epoch(state)
-    previous_epoch_start_slot = compute_start_slot_at_epoch(previous_epoch)
-
-    # Mark stale headers as unconfirmed
-    for slot in range(previous_epoch_start_slot, previous_epoch_start_slot + SLOTS_PER_EPOCH):
-        buffer_index = slot % SHARD_STATE_MEMORY_SLOTS
-        for shard_index in range(len(state.shard_buffer[buffer_index])):
-            committee_work = state.shard_buffer[buffer_index][shard_index]
-            if committee_work.status.selector == SHARD_WORK_PENDING:
-                winning_header = max(committee_work.status.value, key=lambda header: header.weight)
-                if winning_header.attested.commitment == DataCommitment():
-                    committee_work.status.change(selector=SHARD_WORK_UNCONFIRMED, value=None)
-                else:
-                    committee_work.status.change(selector=SHARD_WORK_CONFIRMED, value=winning_header.attested)
-```
+        # Verify that blocks since the last intermediate block have been included
+        blocks_chunked = [block_to_field_elements(ssz_serialize(block)) for block in state.blocks_since_intermediate_block]
```

Comment:
I feel like this is going to be a significant source of complexity. Could even just leave it out at first!

Reply:
Yes, it will not provide any benefits until we can fraud proof beacon and execution blocks. My thinking is currently to spec it out (in order to think through the feasibility and have it "ready" for future deployment), but disable it in the first hard fork.